src/metatrain/utils/additive/composition.py

```diff
@@ -185,19 +204,23 @@ def train_model(
                                     dtype=composition_features.dtype,
                                     device=composition_features.device,
                                 ),
-                                composition_features.T @ targets,
+                                composition_features.T @ all_targets,
                             ).to(self.weights.dtype)
                         )
                         break
                     except torch._C._LinAlgError:
                         regularizer *= 10.0
 
     def restart(self, dataset_info: DatasetInfo) -> "CompositionModel":
-        for target_info in dataset_info.targets.values():
-            if not self.is_valid_target(target_info):
+        """Restart the model with a new dataset info.
+
+        :param dataset_info: New dataset information to be used.
+        """
+        for target_name, target_info in dataset_info.targets.items():
+            if not self.is_valid_target(target_name, target_info):
                 raise ValueError(
-                    f"Composition model does not support target quantity "
-                    f"{target_info.quantity}. This is an architecture bug. "
+                    f"Composition model does not support target "
+                    f"{target_name}. This is an architecture bug. "
```

Comment:
Why the lack of support of the target is an architecture bug? If would not call it a bug, in fact, but maybe it’s just a copy-paste glitch